### PR TITLE
docs(ic): updated Unique ID Push Open API specification to include upstream visibility (connect-to-child)

### PR DESCRIPTION
**Pull request by markuskeidl**
This PR is based on https://github.com/catenax-ng/eclipse-tractusx.github.io/pull/60 and requires a merge of this PR first.

There are still two open issues regarding this change:

- Should the extension of the Unique ID Push API with upstream visibility be integrated into IC which means that the Open API specification in IC must be updated. Or should we add this update as a separate Open API specification only in the Traceabilty KIT. 
- Should there be two EDC assets, one for each method (connect-to-parent, connect-to-child), or should be there be only one EDC asset through which both API calls are routed?

Depending on the answer to the above two questions, different parts of the IC and Trace KIT must be updated.

### CODE REVIEW

**Review by nirschlk — COMMENTED** [review]
looks define to me.

**Comment by eckardg** [comment]
@markuskeidl : I recommend to have two assets and two API specifications.
- two assets: connect-to-parent will be called by the supplier, connect-to-child will be called by the customer; it can be necessary to add different policies
- two API specifications: this makes it easier to refer to the specifications in different KITs and standards

**Comment by arnoweiss** [comment]
Sounds to me like it should be two separate EDC-assets. If it's one and both endpoints are optional, there's no way for a Consumer to know what the API behind the asset will actually satisfy. If it's two, this will be visible in the catalog already.

**Comment by eckardg** [comment]
looks fine for me. The body of the connect-to-child still has to be updated to correspond to the aspect model. I assume we will do this later?

File: openApi/traceability/unique-id-push-trace-extension.yaml [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        - listOfItems
+      properties:
+        information:
+          type: string
+          maxLength: 1000
+          example: "Update of serialized parts, batches, and JIS parts for which digital twins were created."
+        listOfItems:
+          type: array
+          items:
+            oneOf:
```
Comment (arnoweiss) [inline openApi/traceability/unique-id-push-trace-extension.yaml:132]: How will a receiver of this message know what of the three options to parse each payload into?

**Review by arnoweiss — APPROVED** [review]
Thanks for preparing this - it's fine overall. I'd appreciate help understand the parsing logic on the `listOfItems` property

**Review by nirschlk — APPROVED** [review]
fine for me.

File: openApi/traceability/unique-id-push-trace-extension.yaml [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+      description: The classification defines the type of the Unique ID push notification.
+    PartItem:
+      type: object
+      required:
+        - manufacturerId
+        - manufacturerPartId
+        - catenaXId
+      properties:
+        manufacturerId:
+          type: string
```
Comment (DrDanielMetz) [inline openApi/traceability/unique-id-push-trace-extension.yaml:143]: urn_samm_io.catenax.shared.business_partner_number_2.0.0_BpnlTrait

could be used.

File: openApi/traceability/unique-id-push-trace-extension.yaml [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+              type: string
+              example: "OEM-A-F8LM95T92WJ9KNDD3HA5P"
+              description: A number identifying the just-in-sequence- part's destination parent part. The parent part is typically known upfront to the supplier for just-in-sequence parts.
+        - $ref: '#/components/schemas/CatenaXIdItem'
+    CatenaXIdItem:
+      type: object
+      required:
+        - catenaXId
+      properties:
+        catenaXId:
```
Comment (DrDanielMetz) [inline openApi/traceability/unique-id-push-trace-extension.yaml:203]: urn_samm_io.catenax.shared.uuid_2.0.0_UuidV4Trait

could be used

File: openApi/traceability/unique-id-push-trace-extension.yaml (hunk 1 above)
Comment (markuskeidl) [reply openApi/traceability/unique-id-push-trace-extension.yaml:106]: With three options you mean serial part, batch or JIS part? You can see which type of part it is (in the list) using the local IDs. For serialized part, it's partInstanceId, for batches it's batchID and so on.

File: openApi/traceability/unique-id-push-trace-extension.yaml (hunk 2 above)
Comment (markuskeidl) [reply openApi/traceability/unique-id-push-trace-extension.yaml:143]: I already updated that locally. The next version of the PR will use this type.

File: openApi/traceability/unique-id-push-trace-extension.yaml (hunk 3 above)
Comment (markuskeidl) [reply openApi/traceability/unique-id-push-trace-extension.yaml:203]: I already updated that locally. The next version of the PR will use this type.

**Review by DrDanielMetz — CHANGES_REQUESTED** [review]

File: openApi/industrycore/unique-id-push_2-1-0.yaml [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+      - unit:day
+    urn_samm_io.catenax.shared.quantity_2.0.0_QuantityValueCharacteristic:
+      type: number
+      description: The quantity value associated with the unit expressed as float.
+    # -----------------------------------------
+    # Schema: urn:samm:io.catenax.shared.message_header:3.0.0#MessageHeaderAspect
+    # -----------------------------------------
+    urn_samm_io.catenax.shared.message_header_3.0.0_ContextCharacteristic:
+      type: string
+      description: Defining a string value for the context
```
[the remark this answers is not in the document]
Comment (markuskeidl) [reply openApi/industrycore/unique-id-push_2-1-0.yaml:297]: I don't want to do that. The schema for ContextCharacteristic is from the message header standard aspect model. I do not want to change this definition, ideally, if possible, I would only reference this schema externally anyway.

[the remark this answers is not in the document]
Comment (DrDanielMetz) [reply openApi/industrycore/unique-id-push_2-1-0.yaml:297]: Ok, it is fine for me - personally.

I am just wondering how someone could implement this API correctly if it is unclear which values are possible.
The pattern `<domain>-<subdomain>-<object>:<[major] version>` is not sufficient to answer this as there is also no place where the values for domain, subdomain, and object are described.

So, at the end, you simply need to know it - from somewhere.

With regard to the standard header, it is a general question where such details - which are necessary for a working implementation - are to be defined. I could not answer this question today.

**Review by DrDanielMetz — APPROVED** [review]

**Pull request merged**